### PR TITLE
Add some documentation on how to use pod spec patches

### PR DESCRIPTION
**Pull request by triarius**
I've added a section in the README to dump some knowledge of how to use PodSpecPatch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,144 @@ If you have a multi-line `command`, specifying the `args` as well could lead to
...
+It will then apply the patch specified in its config file, which is derived from the value in the helm installation.
+This can replace much of the functionality of some of the other fields in the plugin, like `gitEnvFrom`.
+
+#### Eliminate `gitEnvFrom`
+Here's an example demonstrating how one would eliminate the need to specify `gitEnvFrom` from every step, but still checkout private repositories.
+
+First, deploy the helm chart with a `values.yaml` file.
+```yaml
+# values.yaml
+agentStackSecret: <name of predefend secrets for k8s>
```
Comment (DrJosh9000) [inline README.md:163]:
```suggestion
agentStackSecret: <name of predefined secrets for k8s>
```

File: README.md [hunk 2]
```diff
@@ -148,6 +148,144 @@ If you have a multi-line `command`, specifying the `args` as well could lead to
...
+      - secretRef:
+          name: git-checkout # <---- this is the same secret name you would have put in `gitEnvFrom` in the kubernetes plugin
+```
+You may use the `-f` or `--values` arguments to `helm upgrade` to specify a `values.yaml` file.
+```shell
+helm upgrade --install agent-stack-k8s oci://ghcr.io/buildkite/helm/agent-stack-k8s \
+    --create-namespace \
+    --namespace buildkite \
+    --values values.yaml
+    --
```
Comment (DrJosh9000) [inline README.md:179]: ?

```suggestion
```

**Review by DrJosh9000 — APPROVED** [review]
LGTM with two nits

File: README.md [hunk 3]
```diff
@@ -148,6 +148,144 @@ If you have a multi-line `command`, specifying the `args` as well could lead to
...
```
Comment (triarius) [inline README.md:179]:
```suggestion
    --version <agent-stack-k8s version>
```

File: README.md [hunk 4]
```diff
@@ -148,6 +148,144 @@ If you have a multi-line `command`, specifying the `args` as well could lead to
...
+      envFrom:
+      - secretRef:
+          name: git-checkout # <---- this is the same secret name you would have put in `gitEnvFrom` in the kubernetes plugin
+```
+You may use the `-f` or `--values` arguments to `helm upgrade` to specify a `values.yaml` file.
+```shell
+helm upgrade --install agent-stack-k8s oci://ghcr.io/buildkite/helm/agent-stack-k8s \
+    --create-namespace \
+    --namespace buildkite \
+    --values values.yaml
```
Comment (triarius) [inline README.md:178]:
```suggestion
    --values values.yaml \
```

**Pull request merged**